File: Data-Structure-and-Algorithm--main/Data-Structure-and-Algorithm--main/Unit-04/JS-1(Variables + Data types Mathematical &Comparison Operator)/DSA Weekly Contest 2/Recursive Bubble sort.js
/*Description

You are given an array ofNunsorted numbers. Your task is to write BUBBLE SORT using RECURSION such that numbers present in the array get sorted.

USING ANY OTHER ALGORITHM OR USING SORT FUNCTION WOULD LEAD TO DISQUALIFICATION.




Input
Input Format

First line of input contains N

Second line of input contains N numbers

Constraints

N < 500


Output
Output the numbers given after sorting it in increasing order


Sample Input 1 

5
3 5 0 9 8
Sample Output 1

0 3 5 8 9*/
function findit(array, n) {
  if (n === 0) return;
  for (let i = 0; i < n - 1; i++) {
    if (array[i] > array[i + 1]) {
      [array[i], array[i + 1]] = [array[i + 1], array[i]];
    }
  }
  findit(array, n - 1);
}
function runProgram(input) {
  input = input.trim().split("\n");
  let n = +input[0];
  let array = input[1].trim().split(" ").map(Number);
  findit(array, n);
  console.log(array.join(" "));
}
if (process.env.USERNAME === "Talib") {
  runProgram(``);
} else {
  process.stdin.resume();
  process.stdin.setEncoding("ascii");
  let read = "";
  process.stdin.on("data", function (input) {
    read += input;
  });
  process.stdin.on("end", function () {
    read = read.replace(/\n$/, "");
    read = read.replace(/\n$/, "");
    runProgram(read);
  });
  process.on("SIGINT", function () {
    read = read.replace(/\n$/, "");
    runProgram(read);
    process.exit(0);
  });
}

File: JAVA/Recursive Bubble sort.java
/*
Description

You are given an array ofNunsorted numbers. Your task is to write BUBBLE SORT using RECURSION such that numbers present in the array get sorted.

USING ANY OTHER ALGORITHM OR USING SORT FUNCTION WOULD LEAD TO DISQUALIFICATION.




Input
Input Format

First line of input contains N

Second line of input contains N numbers

Constraints

N < 500


Output
Output the numbers given after sorting it in increasing order


Sample Input 1 

5
3 5 0 9 8
Sample Output 1

0 3 5 8 9*/
import java.util.*;
import java.lang.*;
import java.io.*;
class Main{
    public static void bubblesort(int [] arr, int n){
        if( n==arr.length-2) return;
        for (int j= 0; j<arr.length-n-1;j++){
            if(arr[j]>arr[j+1]){
                int temp = arr[j];
                arr[j] = arr[j+1];
                arr[j+1] = temp;
            }
        }
        bubblesort(arr,n+1);
    }
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        int num = sc.nextInt();
        int [] arr = new int[num];
        for ( int i = 0; i<num; i++){
            arr[i] = sc.nextInt(); }
            bubblesort(arr,0);
            for ( int i = 0; i<num; i++){
                System.out.print(arr[i] + " ");
            }
        }
    }
